### PR TITLE
Transcript for [[nodiscard]] video cr

### PR DESCRIPTION
**Pull request by zacklj89**
Transcription for #9 👍 

I tried following the instructions best I could. Please let me know if I messed anything up!

### CODE REVIEW

File: transcripts/2022/2022-08-25/edited.vtt [hunk 1]
```diff
@@ -0,0 +1,6035 @@
...
+what does the STL use it for,
+
+00:00:39.750 --> 00:00:42.850
+and what do we want to do in this PR?
+
+00:00:42.850 --> 00:00:44.929
+Umm, so if you haven't heard of nodiscard
+
+00:00:44.929 --> 00:00:46.509
+r attributes in general, umm
```
Comment (zacklj89) [inline transcripts/2022/2022-08-25/edited.vtt:61]: or

File: transcripts/2022/2022-08-25/edited.vtt [hunk 2]
```diff
@@ -0,0 +1,6035 @@
+WEBVTT
+
+NOTE duration:"01:16:50"
+
+NOTE recognizability:0.763
```
Comment (fsb4000) [inline transcripts/2022/2022-08-25/edited.vtt:5]:
```suggestion
```

File: transcripts/2022/2022-08-25/edited.vtt [hunk 3]
```diff
@@ -0,0 +1,6035 @@
...
+and then just drop it on the floor,
+
+01:10:21.870 --> 01:10:23.556
+you never joined or detached it.
+
+01:10:23.560 --> 01:10:24.973
+Well, that's bad.
+
+01:10:24.973 --> 01:10:28.546
+It's going to terminate with J thread.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:5435]:
```suggestion
It's going to terminate. With jthread
```

File: transcripts/2022/2022-08-25/edited.vtt [hunk 4]
```diff
@@ -0,0 +1,6035 @@
...
+
+01:10:28.546 --> 01:10:29.670
+That's interesting.
+
+01:10:29.670 --> 01:10:32.764
+Do we mark J thread ctor as
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:5441]:
```suggestion
Do we mark jthread ctor as
```

File: transcripts/2022/2022-08-25/edited.vtt [hunk 5]
```diff
@@ -0,0 +1,6035 @@
...
+
+01:10:32.770 --> 01:10:34.660
+nodiscard?
+
+01:10:34.660 --> 01:10:36.585
+It is unusual to construct a J
+
+01:10:36.585 --> 01:10:38.609
+thread and then do nothing with it.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:5450]:
```suggestion
01:10:34.660 --> 01:10:36.585
It is unusual to construct a jthread 

01:10:36.585 --> 01:10:38.609
and then do nothing with it.
```

File: transcripts/2022/2022-08-25/edited.vtt [hunk 6]
```diff
@@ -0,0 +1,6035 @@
...
+
+01:10:38.610 --> 01:10:39.340
+Let's see.
+
+01:10:43.720 --> 01:10:46.933
+J thread. OK, we did in fact mark J
+
+01:10:46.933 --> 01:10:48.150
+thread ctor as nodiscard ctor.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:5459]:
```suggestion
01:10:43.720 --> 01:10:46.933
jthread. OK, we did in fact mark jthread 

01:10:46.933 --> 01:10:48.150
ctor as nodiscard ctor.
```

**Review by AlexGuteniev — CHANGES_REQUESTED** [review]

File: transcripts/2022/2022-08-25/edited.vtt [hunk 7]
```diff
@@ -0,0 +1,6033 @@
...
+so let's change this.
+
+00:05:08.520 --> 00:05:13.220
+Let's have a couple containers.
+
+00:05:13.220 --> 00:05:16.800
+Let's have the be 10, 20, 30.
+
+00:05:16.800 --> 00:05:22.037
+vector<int> two = {40, 50, 60}.
```
Comment (fsb4000) [inline transcripts/2022/2022-08-25/edited.vtt:470]:
```suggestion
vector&lt;int&gt; two = {40, 50, 60}.
```

File: transcripts/2022/2022-08-25/edited.vtt [hunk 8]
```diff
@@ -0,0 +1,6033 @@
...
+communicate information to the compiler.
+
+00:09:47.470 --> 00:09:49.054
+The compiler just sees a function
+
+00:09:49.054 --> 00:09:50.450
+call here -- operator not equal.
+
+00:09:50.450 --> 00:09:51.700
+For a vector<int> iterator,
```
Comment (fsb4000) [inline transcripts/2022/2022-08-25/edited.vtt:844]:
```suggestion
For a vector&lt;int&gt; iterator,
```

**Review by fsb4000 — CHANGES_REQUESTED** [review]
Nicole changed my `<` and `>` to `&lt;` and `&gt;` so I think we need to replace them.

**Comment by strega-nil-ms** [comment]
I've sent it off to Lisa :)

File: transcripts/2022/2022-08-25/edited.vtt [hunk 9]
```diff
@@ -0,0 +1,6014 @@
...
+We got a bool,
+
+00:08:10.964 --> 00:08:12.160
+then we drop it on the floor,
+
+00:08:12.160 --> 00:08:16.144
+and the result of this is only j.
+
+00:08:16.150 --> 00:08:19.839
+So what we really wanted was operator&&.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:707]: `&amp;&amp;`

File: transcripts/2022/2022-08-25/edited.vtt [hunk 10]
```diff
@@ -0,0 +1,6014 @@
...
+we would
+
+00:08:59.996 --> 00:09:03.710
+end as v2 ends, so it would seem to work fine.
+
+00:09:03.710 --> 00:09:05.774
+But if v1 does not have enough elements,
+
+00:09:05.780 --> 00:09:07.772
+the && will work properly where
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:776]: `&amp;&amp;`

File: transcripts/2022/2022-08-25/edited.vtt [hunk 11]
```diff
@@ -0,0 +1,6014 @@
...
+So usage of != is rewritten
+
+00:11:31.850 --> 00:11:35.258
+to ==, and with
+
+00:11:35.258 --> 00:11:38.690
+&lt;=&gt; any usage of
+
+00:11:38.690 --> 00:11:43.360
+&lt;, &gt;, &lt;=, &gt;= is rewritten to <=>.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:989]: escape

File: transcripts/2022/2022-08-25/edited.vtt [hunk 12]
```diff
@@ -0,0 +1,6014 @@
...
+
+00:11:45.820 --> 00:11:48.780
+Umm the there's articles, uh,
+
+00:11:48.780 --> 00:11:52.248
+that will explain why == and
+
+00:11:52.248 --> 00:11:54.560
+<=> are treated differently.
```
Comment (AlexGuteniev) [inline transcripts/2022/2022-08-25/edited.vtt:998]: escape

**Pull request merged**